clustering/jgroups/WFLY-20444_Upgrade_JGroups_AWS_to_3_0_1_Final.adoc

```diff
@@ -0,0 +1,119 @@
+---
+categories:
+  - clustering
+  - cloud
+stability-level: default
+issue: https://github.com/wildfly/wildfly-proposals/issues/690
+feature-team:
+ developer: rhusar
+ sme:
+  - pferraro
+ outside-perspective:
+  - istraka
+promotes:
+promoted-by:
+---
+= WFLY-20444 Upgrade JGroups AWS to 3.0.1.Final and software.amazon.awssdk to 2.30.32
+:author:            Radoslav Husar
+:email:             rhusar@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Upgrade the JGroups AWS to 3.0.1.Final and software.amazon.awssdk to 2.30.32.
+See proposal for WFLY-8770 for initial integration motivation.
+
+=== User Stories
+
+Users want to configure a discovery protocol in AWS environment using S3 as a cluster information store
```

Comment:
There is a path-style option in the jgroups documentation. However it seems to be deprecated and only possible for buckets created before 2020
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access
- https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/

Reply:
@istraka No, the path‑style addressing was never fully deprecated by them.  AWS’s original plan was to stop supporting it on 30 September 2020, but the plan was changed; path‑style URLs remain supported for buckets created on or before 30 September 2020, and new buckets created after that must use virtual‑hosted style.  No final deprecation date for existing buckets has been announced.

This is the only way to support pre-2020 existing bucket names that do contain a dot in the name that cannot be converted to virtual host style (which was needed for security and efficiency implications).

What this translates to in our protocol is call to software.amazon.awssdk.services.s3.S3BaseClientBuilder#forcePathStyle.

Reply:
Our docs describe this as "Forces the AWS S3 client to use path-style addressing for buckets" which is exactly what the AWS SDK does. The property name is not the clearest, but it's to keep compatibility and prevent users from having to migrate from one prop to another for no benefit.

Please resolve unless there is more to discuss/investigate.

Reply:
I see... resolving.